Draft: Ollama API with Podman Compose

jroddev opened the pull request:
WIP for using podman-compose instead of docker compose with the Ollama API docs

jroddev commented on docs/ai.md:
```diff
@@ -45,10 +47,38 @@ services:
+    ports:
+      - 11434:11434
+    volumes:
+      - ./ollama_v:/root/.ollama:z
```
Permission denied in podman without `:z`

jroddev commented on docs/ai.md:
```diff
@@ -45,10 +47,38 @@ services:
+    volumes:
+      - ./ollama_v:/root/.ollama:z
+    devices:
+      - nvidia.com/gpu=all
```
This might work with docker-compose as well. If so we can probably unify the 2 blocks (`:z` worked with docker-compose)

jroddev replied:
Does not work for `docker`

```
Error response from daemon: could not select device driver "cdi" with capabilities: []
```

castrojo commented:
We could also just put a quadlet in there and tell people to paste it into the right file? I'm thinking, if we're going podman, we should go full podman/quadlet which is what they prefer. 

We should also leave the docker example there too, if someone needs to get something done and they need ollama they shouldn't have to learn podman that same time, so offering both feels great. What do you think?

jroddev commented:
@castrojo I think that makes sense. I suspect that it will still need to run as root, but I'll give it a try and report back.
Also LMStudio AppImage worked out of the box

jroddev commented on docs/ai.md:
```diff
@@ -15,9 +15,84 @@ GPU Acceleration for both Nvidia and AMD are included out of the box and usually
+ContainerName=ollama
+AutoUpdate=yes
+PublishPort=11434:11434
+Volume=./ollama_v:/root/.ollama:z
```
Not sure what the volume path should be in the quadlet

jroddev commented:
OK so the quadlet _kind of works_.
- Starts fine under `--user` with `~/.local/share/systemd/ollama.container`
- nvidia gpu is working (without `sudo`, `--system`)
- I still needed to `brew install ollama` on the host to get the cli frontend (or you could exec into the container)
- ollama is picking up existing models from my system (I think from `~/.ollama`). I'm not sure how since it's not mounted into the container. Maybe the frontend is doing it?

sykoram commented:
Also [here](https://github.com/ublue-os/bluefin/blob/94d23d1e41c5fb092db6eb040e659b95aae6f805/just/bluefin-tools.just#L59) is how the Quadlet unit was being created when running (now removed) `ujust ollama`.

jroddev commented:
@sykoram thanks for sharing.
My quadlet was blocked on this podman issue https://github.com/containers/podman/issues/19338
which has now been merged but I haven't had a chance to test it

Though that might be irrelevant as you pointed out the `ujust` quadlet was using `AddDevice=nvidia.com/gpu=all` and I believe that worked fine.

jroddev commented:
The fix either didn't work or I haven't gotten the version with the patch yet.
Either way I think this page has enough options that we don't need the podman-compose version - especially while it requires `sudo` for nvidia gpu support.

This PR now only adds the Quadlet section because I think that could still be a nice way of doing it.

@sykoram I took your suggestion and switched to using `AddDevice=nvidia.com/gpu=all` . I think it reads nicer than `Device=/dev/nvidia*:ro`. Feel free to make other suggestions.

@castrojo What are your thoughts on this PR? The docs have been updated since I created it and now has information about **Podman Desktop** and **Ramalama** which look to be the preferred options. I've also switched to mostly using LMStudio myself.
Happy for this to be merged or closed.

castrojo commented:
I'll look tomorrow but I think we should keep the ollama sections, lotsd of people still use it.

Over the past week I've been playing a lot with ramalama and it solves lots of problems for us, the cli tool still consumes ollama models from them, but I _love_ that the built in systemd integration is just a flag, the quadlet part gets totally automated, so I think it's a great default. 

There's so much open source AI tooling though, so I think having a few options in the docs for folks makes sense, I'll take a look and merge appropriately!

jroddev commented:
Great suggestions. I have applied the changes and tested Aurora + Nvidia GPU.
I don't have an AMD GPU to test with though.

Having the AMD devices listed in my `.container` file didn't cause any issues though.
```
AddDevice=/dev/dri
AddDevice=/dev/kfd
```

jroddev commented on docs/ai.md:
```diff
@@ -56,9 +56,70 @@ For light chatbot usage we recommend that users [install Alpaca](https://flathub
+Since Alpaca doesn't expose any API, if you need other applications than Alpaca to interact with your ollama instance (for example an IDE) you should consider installing it in a [container](https://hub.docker.com/r/ollama/ollama).
 
-To do so, first configure docker to use the nvidia drivers (that come preinstalled with Bluefin) with:
+### Quadlet (recommended)
```
```
Expected: 1; Actual: 0; Below
```
I don't know what Codacy wants here.

sykoram replied:
Found a [link to the Codacy website for this issue](https://app.codacy.com/gh/ublue-os/bluefin-docs/pull-requests/93/issues#issue-e8a3c8c709f36995d7e33a973d2a5ff3), but there isn't much information and the suggested fix looks exactly the same

sykoram commented:
Since the Quadlet unit file is quite long now, it would be great if someone looks at it and decides which lines are necessary. Unfortunately, I don't have the knowledge for that.

Apart from that, it looks good in my opinion
